x-pack/plugins/apm/common/custom_link/custom_link.test.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { extractTemplateVariableNames, getEncodedCustomLinkUrl } from '.';
+import { Transaction } from '../../typings/es_schemas/ui/transaction';
+
+describe('Custom link', () => {
+  describe('extractTemplateVariableNames', () => {
+    it('extracts variable names from a simple template', () => {
+      const url = 'https://example.com/{{name}}/details';
+      const result = extractTemplateVariableNames(url);
+      expect(result).toEqual(['name']);
+    });
+
+    it('handles multiple occurrences of the same variable', () => {
+      const url = 'https://example.com/{{name}}/details/{{name}}';
+      const result = extractTemplateVariableNames(url);
+      expect(result).toEqual(['name', 'name']);
+    });
```

Comment:
I don't think it should return duplicate values here

```suggestion
    it('handles multiple occurrences of the same variable', () => {
      const url = 'https://example.com/{{name}}/details/{{name}}';
      const result = extractTemplateVariableNames(url);
      expect(result).toEqual(['name']);
    });
```